unity-renderer/Assets/Scripts/MainScripts/DCL/UserProfile/UserProfileController.cs

```diff
@@ -122,8 +123,10 @@ public void RemoveUserProfileFromCatalog(UserProfile userProfile)
 
     public void ClearProfilesCatalog() { userProfilesCatalog?.Clear(); }
 
-    public UniTask<UserProfile> RequestFullUserProfileAsync(string userId)
+    public UniTask<UserProfile> RequestFullUserProfileAsync(string userId, CancellationToken cancellationToken = default)
     {
+        cancellationToken.ThrowIfCancellationRequested();
+
         // TODO: the renderer should not alter the userId nor ethAddress
         // There is a problem in case management of addresses that we cannot handle from renderer only
         userId = userId?.ToLower();
```

Comment:
Try to avoid allocations: just pass "StringComparer.OrdinalIgnoreCase" to the dictionary's constructor